### PR TITLE
Testing Game Check and Checkmate

### PR DESCRIPTION
**Pull request by msarit**
- Wrote tests for `check?(is_white)` and `checkmate?(is_white)` methods, and also for all sub-methods within these
- Modified code in `game.rb`, `king.rb` and `piece.rb` in order to get passing tests

**NOTE**: @bryanjim from what I understood, your `can_be_blocked?(king)` method only checked locations that were directly adjacent to the **threatened_King**. However, a **blocking_Piece** can move to any location along a valid path from the **threatening_Piece** to the **threatened_King**. I adjusted the code in `piece.rb` accordingly. But PLEASE let me know if I interpreted your code wrong 🙏

### CODE REVIEW

**Comment by bryanjim** [comment]
+1

File: app/models/piece.rb [hunk 1]
```diff
@@ -133,60 +141,131 @@ def d_obs?(x,y)
...
 
+  def vertical_path_to(king)
+    if (self.location_x == king.location_x) && ((self.location_y - king.location_y).abs > 1)
+      return true
+    else
+      return false
+    end
+  end
 
-  def move_to!(new_x,new_y)
```
Comment (Linda091) [inline app/models/piece.rb]: @msarit did you remove the move_to! method? if so, can I ask why?

Comment (msarit) [reply app/models/piece.rb]: Oh sorry I didnt see this 😄 I didn't remove it... for some reason there were TWO move_to! methods. One of them was from when we first wrote it, and the other (which I kept) was more fleshed out with tests for valid_move? etc

At least that's what I thought I did lol. I'll double-check, but there is an *updated* move_to! there 👍

Comment (Linda091) [reply app/models/piece.rb]: Thanks! @msarit LEt me know and I'll +1 after :)

**Comment by Linda091** [comment]
+1

**Pull request merged**